elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/FieldHelper.java

```diff
@@ -101,6 +101,76 @@ static boolean isCalendar(EntityIndexBinding indexBinding, String fieldName) {
 		return Calendar.class.isAssignableFrom( propertyClass );
 	}
 
+	static boolean isInstant(EntityIndexBinding indexBinding, String fieldName) {
+		Class<?> propertyClass = getPropertyClass( indexBinding, fieldName );
```

Comment:
It feels to me that a method returning an enum representing the actual type would make much more sense (and be more concise), especially given how those methods are used, but I didn't want to change the way it was done. I'll give it a try.